### PR TITLE
 refactor: Remove usage of `lazy_static!` in `core` and `once_cell`

### PR DESCRIPTION
**Pull request by pmnoxx**
Let's remove usage of `lazy_static!` for `/core`. This will make it easier to review.

Part of #5321

### CODE REVIEW

File: core/store/src/db.rs [hunk 1]
```diff
@@ -193,54 +194,51 @@ impl DBCol {
...
-}
+
+pub static IS_COL_RC: Lazy<Vec<bool>> = Lazy::new(|| {
+    let mut col_rc = vec![false; NUM_COLS];
+    col_rc[DBCol::ColState as usize] = true;
+    col_rc[DBCol::ColTransactions as usize] = true;
+    col_rc[DBCol::ColReceipts as usize] = true;
+    col_rc[DBCol::ColReceiptIdToShardId as usize] = true;
+    col_rc
+});
```
Comment (matklad) [inline core/store/src/db.rs:241]: I'd maybe go directly to const array -- seems easier to change the code once rather than twice.

Comment (pmnoxx) [reply core/store/src/db.rs:241]: @matklad I don't mind doing it now. 
However, I need to investigate on what's the cleanest way to use const array. I haven't seen them used much in our codebase. If you can provide me an example then I would do it right now.

Comment (pmnoxx) [reply core/store/src/db.rs:241]: @matklad What do you mean by const array, do you mean something like

```
static NUMBERS: &'static [i32] = &[0,0,0,0,0,1,1,1,0,0,1,1,1,1,0,0];
```
I guess you had something else in mind.

Comment (pmnoxx) [reply core/store/src/db.rs:241]: @matklad Any ideas?

Comment (matklad) [reply core/store/src/db.rs:241]: THe following seems to work: 

```rust
pub const SKIP_COL_GC: [bool; NUM_COLS] = {
    let mut col_gc = [false; NUM_COLS];
    // A node may never restarted
    col_gc[DBCol::ColStateHeaders as usize] = true;
    // True until #2515
    col_gc[DBCol::ColStateParts as usize] = true;
    col_gc
};
```

**Comment by pmnoxx** [comment]
@matklad Can you take a look again?

**Comment by pmnoxx** [comment]
@matklad I removed once_cell usage from `store`.

**Review by bowenwang1996 — APPROVED** [review]

**Pull request merged**